Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -118,155 +119,133 @@ public override void HandleComponentState(ComponentState? curState, ComponentSta
         {
-            if (!SupportsDamageClass(damageGroup))
-            {
-                return 0;
-            }
-
-            var damage = 0;
-
-            foreach (var type in damageGroup.DamageTypes)
-            {
-                damage += GetDamage(type);
-            }
-
-            return damage;
+            return DamagePerGroup.GetValueOrDefault(group);
         }
-
-        public bool TryGetDamage(DamageGroupPrototype damageGroup, out int damage)
+        public bool TryGetDamage(DamageGroupPrototype group, out int damage)
         {
-            if (!SupportsDamageClass(damageGroup))
-            {
-                damage = 0;
-                return false;
-            }
-
-            damage = GetDamage(damageGroup);
-            return true;
+            return DamagePerGroup.TryGetValue(group, out damage);
         }
-
-        /// <summary>
-        ///     Attempts to set the damage value for the given <see cref="DamageTypePrototype"/>.
-        /// </summary>
-        /// <returns>
-        ///     True if successful, false if this container does not support that type.
-        /// </returns>
-        public bool TrySetDamage(DamageTypePrototype type, int newValue)
+        public bool IsApplicableDamageGroup(DamageGroupPrototype group)
         {
-            if (newValue < 0)
-            {
-                return false;
-            }
-
-            if (SupportedTypes.Contains(type))
-            {
-                var old = _damageList[type] = newValue;
-                _damageList[type] = newValue;
-
-                var delta = newValue - old;
-                var datum = new DamageChangeData(type, newValue, delta);
-                var data = new List<DamageChangeData> { datum };
-
-                OnHealthChanged(data);
-
-                return true;
-            }
-
-            return false;
+            return ApplicableDamageGroups.Contains(group);
+        }
+        public bool IsSupportedDamageGroup(DamageGroupPrototype group)
+        {
+            return SupportedDamageGroups.Contains(group);
         }
+        public bool IsSupportedDamageType(DamageTypePrototype type)
+        {
+            return SupportedDamageTypes.Contains(type);
+        }
+        
+
 
-        public void Heal(DamageTypePrototype type)
+        public void SetGroupDamage(int newValue, DamageGroupPrototype group)
         {
-            SetDamage(type, 0);
+            foreach (var type in group.DamageTypes)
+            {
+                SetDamage(type, newValue);
+            }
         }
 
-        public void Heal()
+        public void SetAllDamage(int newValue)
         {
-            foreach (var type in SupportedTypes)
+            foreach (var type in SupportedDamageTypes)
             {
-                Heal(type);
+                SetDamage(type, newValue);
             }
         }
 
+        // TODO QUESTION both source and extraParams are unused here. Should they be removed, or will they have use in
```

Comment:
Should these parameters just be removed?

Reply:
remove them, i dont really see any reason they need to exist, and they can be readded later if they end up being useful